llvm/test/Transforms/CorrelatedValuePropagation/sub.ll

```diff
@@ -50,10 +50,9 @@ exit:
 define void @test2(i32 %a) {
 ; CHECK-LABEL: @test2(
 ; CHECK-NEXT:  entry:
-; CHECK-NEXT:    [[CMP:%.*]] = icmp ugt i32 [[A:%.*]], -1
-; CHECK-NEXT:    br i1 [[CMP]], label [[BB:%.*]], label [[EXIT:%.*]]
+; CHECK-NEXT:    br i1 false, label [[BB:%.*]], label [[EXIT:%.*]]
```

Comment:
I believe this is a miscompilation.

Reply:
This looks correct to me. It's just that LVI normally doesn't bother handling tautological comparisons at all, and the new implementation happens to handle them.